alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -156,5 +158,18 @@ class AlertingSettings {
             -1L,
             Setting.Property.NodeScope, Setting.Property.Dynamic
         )
+
+        val ALERTING_TRIGGER_MAX_ACTIONS = Setting.intSetting(
+            "plugins.alerting.trigger.max_actions",
+            DEFAULT_ALERTING_TRIGGER_MAX_ACTIONS,
+            -1,
```

Comment:
Why not propose a default max value to promote best practice?

Reply:
> Why not propose a default max value to promote best practice?

Should we ask OpenSearch side give us an optimal default value? Since the user may directly use the default value provided by us and Opensearch should be the most experienced in how much this default value should be set?

Reply:
I believe we should set this to -1 as the default because we do not want actions to stop executing for users who upgrade their cluster. As for best practices, do you have suggestions on how we can come up with a recommended value that we can indicate on our documentation as this is heavily dependent on what their cluster is used for and the size of it.

Reply:
>  we do not want actions to stop executing for users who upgrade their cluster

We can have the limit validation enforced only during this new settings update. Also this limit should act in during the new action updates, limiting any more actions to be added if limit is breached. While during the actual action execution we can still go with the  total number of actions configured. Thoughts?

> recommended value that we can indicate on our documentation

It's hard to enforce that way and more than best practice, I also see this as a resiliency knob for the cluster. We can always start with a safe limit, which is appropriate/rational and does not impact a large set of customers (Such as 1000). 

We have had such additions in the past as well such as OpenSearch 7.x and later versions enforced a limit of 1,000 shards per node via ``cluster.max_shards_per_node setting``. https://docs.aws.amazon.com/opensearch-service/latest/developerguide/sizing-domains.html

Reply:
We need to do the validation by passing in a `Setting.Validator<Int>` object in here. 
To make a validator, follow how [this is done](https://github.com/opensearch-project/OpenSearch/blob/3aef125d0dbf2f29d9bf00b84cb52892a27a5328/server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdSettings.java#L150-L173)